### PR TITLE
Update FileInstallationStore to behave more consistently

### PR DESCRIPTION
**Pull request by seratch**
###  Summary

This pull request improves the following points in the `@slack/oauth` package:

* Change all the write operations to use "sync" methods for atomicity
* Improve the exceptions coming from FileInstallationStore to include more details of the cause
* Add the corresponding tests for the above changes
* Simplify the rest in tests

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: packages/oauth/src/stores/file-store.spec.js [hunk 1]
```diff
@@ -72,10 +72,15 @@ const storedOrgInstallation = {
...
-  let fsMakeDir, fsWriteFile, fsReadFileSync;
+  let fsMakeDir, fsMakeDirSync,
+    fsWriteFile, fsReadFileSync,
+    fsUnlink, fsReaddirSync;
 
   beforeEach(() => {
+    // Note that these sinon stubs affect the `os` package behaviors
+    // in the tests in this file
     fsMakeDir = sinon.stub(fs, 'mkdir').returns({});
+    fsMakeDirSync = sinon.stub(fs, 'mkdirSync').returns({});
```
Comment (seratch) [inline packages/oauth/src/stores/file-store.spec.js:83]: This stub had been missing

File: packages/oauth/src/stores/file-store.spec.js [hunk 2]
```diff
@@ -92,24 +98,36 @@ describe('FileInstallationStore', async () => {
 
   it('should store the latest installation', async () => {
     const installationStore = new FileInstallationStore({ baseDir: os.tmpdir() });
-    const installer = new InstallProvider({ clientId, clientSecret, stateSecret, installationStore });
```
Comment (seratch) [inline packages/oauth/src/stores/file-store.spec.js:95]: Except for the `authorize` tests, we don't need to instantiate `InstallProvider`.  In fact, this test uses only `installer.installationStore`.

File: packages/oauth/src/stores/file-store.ts [hunk 3]
```diff
@@ -105,36 +105,28 @@ export default class FileInstallationStore implements InstallationStore {
...
   }
 }
 
 function writeToFile(filePath: string, data: string): void {
-  fs.writeFile(filePath, data, (err) => {
-    if (err !== null) {
-      throw new Error('There was an error writing to the InstallationStore');
-    }
-  });
+  fs.writeFileSync(filePath, data);
```
Comment (seratch) [inline packages/oauth/src/stores/file-store.ts:127]: This needs to be "sync" to make the write operation atomic

File: packages/oauth/src/stores/file-store.ts [hunk 4]
```diff
@@ -105,36 +105,28 @@ export default class FileInstallationStore implements InstallationStore {
...
 }
 
 function deleteFile(filePath: string): void {
-  fs.unlink(path.resolve(filePath), (err) => {
-    if (err !== null) {
-      throw new Error('Failed to delete installation from FileInstallationStore');
-    }
-  });
+  fs.unlinkSync(path.resolve(filePath));
```
Comment (seratch) [inline packages/oauth/src/stores/file-store.ts:131]: This needs to be "sync" to make the write operation atomic

File: packages/oauth/src/stores/file-store.ts [hunk 5]
```diff
@@ -105,36 +105,28 @@ export default class FileInstallationStore implements InstallationStore {
       filesToDelete = filesToDelete.concat(userFiles);
     }
 
-    try {
-      filesToDelete.map((filePath) => deleteFile(path.resolve(`${installationDir}/${filePath}`)));
-    } catch (err) {
-      throw new Error('Failed to delete installation from FileInstallationStore');
-    }
+    filesToDelete.forEach((filePath) => {
```
Comment (seratch) [inline packages/oauth/src/stores/file-store.ts:108]: The error handling here should be done for each deletion

File: packages/oauth/src/stores/file-store.ts [hunk 6]
```diff
@@ -83,7 +83,7 @@ export default class FileInstallationStore implements InstallationStore {
       }
       return installation;
     } catch (err) {
-      throw new Error('Failed to fetch data from FileInstallationStore');
+      throw new Error(`No installation data found (enterprise_id: ${query.enterpriseId}, team_id: ${query.teamId}, user_id: ${query.userId})`);
```
Comment (seratch) [inline packages/oauth/src/stores/file-store.ts:86]: Updated to have more information

**Review by seratch — COMMENTED** [review]
Revisited more and changed the title of this PR

**Review by misscoded — APPROVED** [review]

File: packages/oauth/src/stores/file-store.ts (hunk 5 above)
Comment (filmaj) [reply packages/oauth/src/stores/file-store.ts:108]: Doesn't this technically change the behaviour compared to the previous version? In the previous version, if `path.resolve` or `fs.unlink` would error, an exception was raised, whereas in this PR's version, no exception is thrown and an error is logged out instead.

**Review by filmaj — COMMENTED** [review]
I have one question around changing the behaviour (logging out instead of throwing) that I'd like clarity on before I am comfortable moving forward. Thanks for adding the new tests though and improving the code overall!

File: packages/oauth/src/stores/file-store.ts (hunk 5 above)
Comment (seratch) [reply packages/oauth/src/stores/file-store.ts:108]: @filmaj Good point. I can revert the logic change here and add only log improvement this time.

Comment (seratch) [reply packages/oauth/src/stores/file-store.ts]: This has been improved

**Review by filmaj — APPROVED** [review]
Nice, thanks! Not sure which approach is better (to throw or just to log) but if we do make a change there we should probably wait for a major version bump?

In any case: :shipit:

**Pull request merged**